jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -199,126 +268,192 @@ def _calc_bg_subset_median(self, reg):
-            self.hub.broadcast(SnackbarMessage(
-                "No data for aperture photometry", color='error', sender=self))
-            return
+    def calculate_photometry(self, dataset=None, aperture=None, background=None,
+                             background_value=None, pixel_area=None, counts_factor=None,
+                             flux_scaling=None, add_to_table=True, update_plots=True):
+        """
+        Calculate aperture photometry given the values set in the plugin or any overrides provided
+        as arguments here (which will temporarily override plugin values for this calculation only).
+
+        Parameters
+        ----------
+        * ``dataset`` : str, optional
+            Dataset to use for photometry.
+        * ``aperture`` : str, optional
+            Subset to use as the aperture.
+        * ``background`` : str, optional
+            Subset to use to calculate the background.
+        * ``background_value`` : float, optional
+            Background to subtract, same unit as data.  Automatically computed if ``background``
+            is set to a subset.
+        * ``pixel_area`` : float, optional
+            Pixel area in arcsec squared, only used if sr in data unit.
+        * ``counts_factor`` : float, optional
+            Factor to convert data unit to counts, in unit of flux/counts.
+        * ``flux_scaling`` : float, optional
+            Same unit as data, used in -2.5 * log(flux / flux_scaling).
+        * ``add_to_table`` : bool, optional
+        * ``update_plots`` : bool, optional
+
+        Returns
+        -------
+        * table row, fit results
+        """
+        if self.multiselect and (dataset is None or aperture is None):  # pragma: no cover
+            raise ValueError("for batch mode, use calculate_batch_photometry")
 
-        data = self._selected_data
-        reg = self._selected_subset
+        if dataset is not None:
+            if dataset not in self.dataset.choices:  # pragma: no cover
+                raise ValueError(f"dataset must be one of {self.dataset.choices}")
+            data = self.dataset._get_dc_item(dataset)
+        else:
+            # we can use the pre-cached value
+            data = self.dataset.selected_dc_item
+
+        if aperture is not None and aperture not in self.aperture.choices:
+            raise ValueError(f"aperture must be one of {self.aperture.choices}")
+        if aperture is not None or dataset is not None:
+            reg = self.aperture._get_spatial_region(subset=aperture if aperture is not None else self.aperture.selected,  # noqa
+                                                    dataset=dataset if dataset is not None else self.dataset.selected)  # noqa
+        else:
+            # use the pre-cached value
+            reg = self.aperture.selected_spatial_region
 
         # Reset last fitted model
         fit_model = None
+        # TODO: remove _fitted_model_name cache?

```

Comment:
FWIW I originally used this because it was already used in model fitting. Since Imviz does not do model fitting, I thought why not just use something that is already there.